Since 2.6.25 class_device_add() and class_device_destroy() has been renamed, 
modified st7554.c accordingly.
Index: slmodem-2.9.11-20080817/drivers/st7554.c
===================================================================
--- slmodem-2.9.11-20080817.orig/drivers/st7554.c	2008-10-04 01:02:35.000000000 +0200
+++ slmodem-2.9.11-20080817/drivers/st7554.c	2008-10-04 00:32:21.000000000 +0200
@@ -220,11 +220,15 @@
 #define CLASS_CREATE(owner, name) class_simple_create(owner, name)
 static struct class_simple *st7554_class;
 #else
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,25)
+#define CLASS_DEVICE_CREATE(class, dev, device, fmt, rest) device_create(class, dev, device, fmt, rest)
+#define CLASS_DEVICE_DESTROY(class, dev) device_destroy(class, dev)
+#else
 #include <linux/moduleparam.h>
-
 #define CLASS_DEVICE_CREATE(class, dev, device, fmt, rest) class_device_create(class, dev, device, fmt, rest)
-#define CLASS_DESTROY(class) class_destroy(class)
 #define CLASS_DEVICE_DESTROY(class, dev) class_device_destroy(class, dev)
+#endif
+#define CLASS_DESTROY(class) class_destroy(class)
 #define CLASS_CREATE(owner, name) class_create(owner, name)
 static struct class *st7554_class;
 #endif
